src/Microsoft.OData.Core/JsonLight/ODataJsonLightInputContext.cs

```diff
@@ -151,6 +151,23 @@ internal Stream Stream
             }
         }
 
+        /// <summary>
+        /// Returns whether to read odata control information without the odata prefix.
+        /// True for OData 4.01 and greater. Settable for OData 4.0
+        /// </summary>
+        internal bool OptionalODataPrefix

```

Comment:
>OptionalODataPrefix [](start = 22, length = 19)

omit or optional ?

Maybe it's a little bit confusing for maintenance.

Reply:
If my understanding is correct optional is fine. Since we are going to support reading both; it is optional. Whereas while writing we support only one thing (omit).

Reply:
Right; see Kanish's comment: this property means that it's optional on reading whether the prefix is there or not, versus writing where we either omit or we don't.

---
In reply to: [279946882](https://github.com/OData/odata.net/pull/1454#discussion_r279946882) [](ancestors = 279946882)

Reply:
xactly.:grinning: 

---
In reply to: [285693803](https://github.com/OData/odata.net/pull/1454#discussion_r285693803) [](ancestors = 285693803)